server/src-lib/Hasura/RQL/DDL/Metadata.hs

```diff
@@ -142,19 +143,20 @@ runClearMetadata _ = do
 
 data ReplaceMetadata
   = ReplaceMetadata
-  { aqTables           :: ![TableMeta]
-  , aqFunctions        :: !(Maybe [QualifiedFunction])
-  , aqRemoteSchemas    :: !(Maybe [TRS.AddRemoteSchemaQuery])
-  , aqQueryCollections :: !(Maybe [DQC.CreateCollection])
-  , aqAllowlist        :: !(Maybe [DQC.CollectionReq])
+  { aqTables              :: ![TableMeta]
+  , aqFunctions           :: !(Maybe [QualifiedFunction])
```

Comment:
I know this isn't new code, but anytime I see `Maybe [Something]`, I want to know (in a comment) the the semantic difference between `Just []` and `Nothing`. If there is no difference then it should probably just be a plain list.

Reply:
I think most of these `Maybe [a]` is to maintain backward compatibility. We want to have backward compatible `replace_metadata` queries so the metadata from v1 (which didn't have `aqFunctions`) still needs to work with later versions.

Reply:
Is there a better way to solve this issue? Write our own parser instead of template haskell?

Reply:
That makes sense, thanks. Perhaps e.g. `Maybe (NonEmpty QualifiedFunction)` is what we want.